sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -793,6 +812,9 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
           initialStateGroupAttrs, data, initialStateDataAttrs, output, timeout,
           hasInitialState, planLater(initialState), planLater(child)
         ) :: Nil
+      case _: FlatMapGroupsInPandasWithState =>
+        throw new UnsupportedOperationException(
```

Comment:
In Scala/Java, flatMapGroupsWithState API uses different physical plan for batch query, which applies some wrap-up with user function to effectively match up with user function in MapGroupsExec.

This seems to be non-trivial to deal with in PySpark, or at least seems to require major work. As there is another API for batch-friendly query (applyInPandas), I think we can leave this to TODO and guide users to use applyInPandas instead for now.

self-comment: file a JIRA ticket and replace the number.

Reply:
Updated.